network/builder/src/builder.rs

```diff
@@ -369,6 +268,12 @@ impl NetworkBuilder {
         );
         self.conn_mgr_reqs_tx = Some(conn_mgr_reqs_tx);
         let pm_conn_mgr_notifs_rx = self.add_connection_event_listener();
+        let connection_reqs_tx = match &self.peer_manager_builder {
```

Comment:
again, same here, the logic is a little confusing since these are /expected/ to be here.

Reply:
Yeah.  These options are really bad hacks to work around (a) needing to set values when the struct is created and (b) the apparent need to drop the object into the tokio.runtime.

There is a slight question (not quite there with the code to make it obvious) if there is benefit to (a) treating all of the builders the same way for symmetry or (b) being smart with special casing.

The latter is more concise and does fewer unnecessary things;  the former is more flexible and imposes fewer handcuffs moving forward.

Reply:
Well, considering the panic below, I think it seems perfectly fine to do the following:
`self.peer_manager_builder.as_ref().expect("...");`